client/disputes/index.tsx

```diff
@@ -226,9 +226,19 @@ export const DisputesList = (): JSX.Element => {
 	const totalRows = disputesSummary.count || 0;
 
 	const rows = disputes.map( ( dispute ) => {
+		const onClickDisputeRow = (
+			e: React.MouseEvent< HTMLAnchorElement >
+		) => {
+			// Use client-side routing to avoid page refresh.
+			e.preventDefault();
+			recordEvent( 'wcpay_disputes_row_action_click' );
+			const history = getHistory();
+			history.push( getDetailsURL( dispute.charge_id, 'transactions' ) );
+		};
 		const clickable = ( children: React.ReactNode ): JSX.Element => (
 			<ClickableCell
 				href={ getDetailsURL( dispute.charge_id, 'transactions' ) }
+				onClick={ onClickDisputeRow }
```

Comment:
In testing, I discovered that `onClick` of `<ClickableCell />` is not fired when using `CMD + click`, therefore the tracks event is not recorded.

This is also the case for deposit table row `CMD + click` events.

This is a minor issue, unsure if it's worth fixing in this PR.

Reply:
> In testing, I discovered that onClick of <ClickableCell /> is not fired when using CMD + click, therefore the tracks event is not recorded.

TIL!

> This is a minor issue, unsure if it's worth fixing in this PR.

Agree it's not core to this PR, but feels like a moderate problem / gap with our components. My take is that we should be able to support cmd-click where practical, and those user actions should be tracked consistently. The part that's confusing me is cmd-click works – the user action happens – but the tracking doesn't. Could be an issue down in woo `Link` component.


https://github.com/woocommerce/woocommerce/blob/a0b72b0546cbf6366defbe7db4e84f692de4e200/packages/js/components/src/link/index.tsx#L37-L46

Can you log an issue for this?

Reply:
Hmm … maybe we could add native tracks ability to components 🤔 - i.e. pass event name and props as component props, and delegate the `recordEvent` to the component.